Fix for Entity Controller setup causing build fails.

rma-kayla opened the pull request:
Refactored and cleaned Entity Controller to address clean runner build fail PR #1507.

- Fixed entity controller integration tests to incorporate unique data per test 
- Added test utility helpers for cleanup and unique entityJson test values
- Cleaned up global variables to remove hard coded param values
- Fixed teardown method to ensure all test entities cleaned up after testing.
- Added check in Entity Controller to fix unused documented query param errors and check for required params

rma-kayla committed:
Refactor and clean Entity Controller due to clean runner build fail (PR1507)

- Fixed EntityControllerTestIT to incorporate unique data per test using test utility helpers.
- Removed hard coded static values
- Fixed teardown to ensure all test entities cleaned up after testsing.
- Added check in EntityController to fix unused documented query param errors

rma-kayla commented:
@rma-bryson and/or @adamkorynta this PR is ready for a review
@MikeNeilson this should fix the build fails from PR #1507

MikeNeilson commented on cwms-data-api/src/main/java/cwms/cda/api/EntityController.java:
```diff
@@ -190,13 +191,27 @@ public void update(@NotNull Context ctx, @NotNull String entityId) {
+
+            if (!officeId.equalsIgnoreCase(entity.getId().getOfficeId())) {
+                ctx.status(HttpServletResponse.SC_BAD_REQUEST);
+                ctx.result("Office ID in query parameter must match the Office ID in the request body.");
```
don't we have "global" entities under the cwms office?

rma-kayla replied:
After discussion with Bryson, we decided the office query parameter is not needed in update method and will be validated in the body.  I removed unnecessary office id checks in the update method.

rma-bryson commented on cwms-data-api/src/main/java/cwms/cda/api/EntityController.java:
```diff
@@ -190,13 +191,27 @@ public void update(@NotNull Context ctx, @NotNull String entityId) {
+
+            if (!entityId.equalsIgnoreCase(entity.getId().getName())) {
+                ctx.status(HttpServletResponse.SC_NOT_FOUND);
+                ctx.result("Entity not found for the given entity-id.");
```
shouldn't this be a similar message and error response code to the mismatched office-id? something like:

""Entity ID in path parameter must match the Entity ID in the request body.""

rma-kayla replied:
updated message to be more consistent.

MikeNeilson replied:
It doesn't look like the change Bryson asked for made it to github.

rma-kayla replied:
@MikeNeilson, it is in next commit.

rma-kayla committed:
Refactored entity update controller and tests.

- Removed office-id query param and redundant checks
- Added missing entity-id validation test
- Added invalid office authorization test
- Improved variable naming
- Fixed failing .replace() logic

rma-kayla commented:
Refactored entity update controller and tests.
- Removed office-id query param and redundant checks, cleaned up error message.
- Added missing entity-id validation test and invalid office authorization test
- Improved variable naming and fixed failing .replace() logic.

MikeNeilson commented on cwms-data-api/src/test/resources/cwms/cda/data/dto/entity_test.json:
I like the naming scheme you've chosen here. It's very explicit as to what's going on.

rma-bryson commented on cwms-data-api/src/main/java/cwms/cda/api/EntityController.java:
Thinking about the match-null-parents flag in the getAll. It doesn't seem right that we would ask for entities with a specific-parent and also get back entities with null-parents. 

I'm thinking we should add some logic here that will switch match-null-parents to false IF the parent-id parameter is provided by the user

rma-bryson replied:
@MikeNeilson - What are your thoughts on that?

MikeNeilson replied:
That make sense to me.

rma-kayla replied:
Added logic to getAll when parent id param is provided, match-null-parents = false.
Added IT to verify logic

rma-kayla committed:
Added logic to getAll when parent id param is provided.

- Added IT to verify logic

rma-bryson reviewed (changes requested):
The version type should be V1

rma-kayla committed:
Replaced JsonV2 with JsonV1 in controller, controllerIT and dto.

rma-kayla commented:
Version in DTO, Controller and Integration tests are updated to JsonV2
